sky/resources.py

```diff
@@ -171,6 +172,7 @@ def __init__(
         self._try_validate_disk_tier()
         self._try_validate_ports()
 
+    @service_catalog.use_default_catalog_if_failed
```

Comment:
Not obvious why a repr func would fetch catalog inside. Comment?

Reply:
Added the comment above. PTAL. Thanks!